discovery-infra/tests/test_discovery_ignition_override.py

```diff
@@ -1,37 +1,159 @@
+        Verify the override was applied
+        """
         # Define new cluster
         cluster_id = cluster().id
+
+        ignition_override = copy.deepcopy(TestDiscoveryIgnition.base_ignition)
+        override_path = "/ignition/test_discovery_ignition_override"
+        ignition_override["storage"]["files"][0]["path"] = override_path
         api_client.patch_cluster_discovery_ignition(cluster_id, ignition_override)
+
         # Generate and download cluster ISO
         self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
         # Boot nodes into ISO
         hosts = node_controller.start_all_nodes()
         # Wait until hosts are discovered and update host roles
         self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
         # Verify override
+        self.validate_ignition_override(hosts, TestDiscoveryIgnition.override_path)
+
+    @pytest.mark.regression
+    def test_discovery_ignition_after_ISO_was_created(self, api_client, node_controller, cluster):
+        """ Verify that we create a new ISO upon discovery igniton override in case one already exists.
+        Download the ISO
+        Create a discovery ignition override
+        Download the ISO again
+        Start the nodes
+        Verify the override was applied
+        """
+        # Define new cluster
+        cluster_id = cluster().id
+
+        ignition_override = copy.deepcopy(TestDiscoveryIgnition.base_ignition)
+        override_path = "/ignition/test_discovery_ignition_after_ISO_was_created"
+        ignition_override["storage"]["files"][0]["path"] = override_path
+
+        # Generate and download cluster ISO
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        # Apply the patch after the ISO was created
+        api_client.patch_cluster_discovery_ignition(cluster_id, ignition_override)
+        # Generate and download cluster ISO again
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+
+        # Boot nodes into ISO
+        hosts = node_controller.start_all_nodes()
+        # Wait until hosts are discovered and update host roles
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
+        # Verify override
+        self.validate_ignition_override(hosts, override_path)
+
+    def validate_ignition_override(self, hosts, file_path, expected_content=TEST_STRING):
         for test_host in hosts.values():
-            file_content = test_host.run_command("cat {}".format(override_path))
-            assert file_content == test_string
+            logging.info("Verifying {} applied the override".format(test_host.name))
+            file_content = test_host.run_command("cat {}".format(file_path))
+            assert file_content == expected_content
 
     @pytest.mark.regression
     def test_discovery_ignition_bad_version(self, api_client, node_controller, cluster):
```

Comment:
This is covered by unit tests ... but :man_shrugging:

Reply:
I'll remove it